test/jdk/com/sun/jdi/TestScaffold.java

```diff
@@ -461,44 +461,64 @@ protected void failure(String str) {
         testFailed = true;
     }
 
+    final List<String> doubleWordArgs = List.of("-cp", "-classpath", "--add-opens", "--class-path",
+            "--upgrade-module-path", "--add-modules", "-d", "--add-exports", "--patch-module", "--module-path");
+
     private ArgInfo parseArgs(String args[]) {
         ArgInfo argInfo = new ArgInfo();
-        String mainWrapper = System.getProperty("main.wrapper");
-        if ("Virtual".equals(mainWrapper)) {
-            argInfo.targetAppCommandLine = TestScaffold.class.getName() + " " + mainWrapper + " ";
-        } else if ("true".equals(System.getProperty("test.enable.preview"))) {
-            // the test specified @enablePreview.
-            argInfo.targetVMArgs += "--enable-preview ";
-        }
-
+        // Parse arguments, like java/j* tools command-line arguments
+        // the first argument not-starting with '-' is treated as a classname
+        // the other arguments are split to targetVMArgs targetAppCommandLine correspondingly
+        // The example of args for line '@run driver Frames2Test -Xss4M' is  '-Xss4M' 'Frames2Targ'.
+        // The result without any wrapper enabled:
+        // argInfo.targetAppCommandLine : Frames2Targ
+        // argInfo.targetVMArgs : -Xss4M
+        // The result with wrapper enabled:
+        // argInfo.targetAppCommandLine : TestScaffold Virtual Frames2Targ
+        // argInfo.targetVMArgs : -Xss4M
```

Comment:
The 2 argInfo comments would be easier to read if indented 2 or 4 spaces (after the //, not before).